### PR TITLE
Offset prober scripts for kafka10

### PR DESCRIPTION
**Pull request by hhakimi53**
Kafka scripts for probing, positioning offsets and reading message from kafka cluster using Kafka 10 api. Script files uses new consumer api (scripts for kafka8 clusater uses old simple consumer and offset meta-data infrastructure which is no longer supported in Kafka 10). For resetting the offsets through the scripts, no other consumer should be assigned to the topic-partition.

### CODE REVIEW

File: tools/kafka/kafka10-offset-info.groovy [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+@Grapes([
+    @Grab('com.google.guava:guava:19.0'),
+    @Grab('log4j:log4j:1.2.17'),    
+	@Grab('org.apache.kafka:kafka-clients:0.10.2.1'),
+	//@Grab('org.slf4j:slf4j-api:1.7.5'),
```
Comment (akhanzode) [inline tools/kafka/kafka10-offset-info.groovy]: remove commented code.

File: tools/kafka/kafka10-offset-info.groovy [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        OffsetAndMetadata commitOffset = consumer.committed(tp)
+        committed = 0
+        if (commitOffset != null) {
+            committed = commitOffset.offset() 
+        }
+
+        long lag = (committed == 0 || committed < startOffset) ? (endOffset - startOffset) :
+            (committed > endOffset) ? 0L : (endOffset - committed)        
+        long nextOffset = consumer.position(tp)
+        //long offset = (committed == 0 || committed < startOffset) ? startOffset : (committed > endOffset) ? endOffset : committed
```
Comment (akhanzode) [inline tools/kafka/kafka10-offset-info.groovy]: remove commented code.

File: tools/kafka/kafka10-message-prober.groovy [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+@Grapes([
+    @Grab('com.google.guava:guava:19.0'),
+    @Grab('log4j:log4j:1.2.17'),    
+	@Grab('org.apache.kafka:kafka-clients:0.10.2.1'),
```
Comment (akhanzode) [inline tools/kafka/kafka10-message-prober.groovy]: indentation?

File: tools/kafka/kafka10-message-prober.groovy [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+    @GrabExclude('javax.mail:mail'),
```
Comment (akhanzode) [inline tools/kafka/kafka10-message-prober.groovy]: you sure you need java mail api and jms?

File: tools/kafka/kafka10-message-prober.groovy [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+    (GROUP_ID_CONFIG): group,
+    (AUTO_OFFSET_RESET_CONFIG): 'earliest',
+    (KEY_DESERIALIZER_CLASS_CONFIG): ByteArrayDeserializer.class.name,
+    (VALUE_DESERIALIZER_CLASS_CONFIG): ByteArrayDeserializer.class.name,
+    (BOOTSTRAP_SERVERS_CONFIG): opts.b,
+    (ENABLE_AUTO_COMMIT_CONFIG): false,
+    (REQUEST_TIMEOUT_MS_CONFIG): (TIMEOUT + 30),
+    (SESSION_TIMEOUT_MS_CONFIG): TIMEOUT,
+    (FETCH_MAX_WAIT_MS_CONFIG): TIMEOUT,
+//    (RECEIVE_BUFFER_CONFIG): (64 << 10), /// commented at preesent - for expiremental purpose
```
Comment (akhanzode) [inline tools/kafka/kafka10-message-prober.groovy]: remove experimental commented code.

File: tools/kafka/kafka10-offset-info.groovy [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+@Grapes([
+    @Grab('com.google.guava:guava:19.0'),
+    @Grab('log4j:log4j:1.2.17'),    
+	@Grab('org.apache.kafka:kafka-clients:0.10.2.1'),
```
Comment (akhanzode) [inline tools/kafka/kafka10-offset-info.groovy]: indent and same comment for dependency.

File: tools/kafka/kafka10-message-prober.groovy (hunk 4 above)
Comment (hhakimi53) [inline tools/kafka/kafka10-message-prober.groovy]: no, that's left over. will clean it up

File: tools/kafka/kafka10-message-prober.groovy (hunk 5 above)
Comment (hhakimi53) [inline tools/kafka/kafka10-message-prober.groovy]: ok

**Pull request merged**